docs/propagation.md

```diff
@@ -0,0 +1,215 @@
+# Propagation
+
+## Overview
+
+Sharding propagation uses the user-specified shardings to infer the unspecified shardings of tensors (or specific dimension of tensors). It traverses the data flow (use-def chains) of the computation graph in both directions until a fixed point is reached, i.e., the sharding can no longer change without undoing previous sharding decisions.
+
+Propagation can be decomposed into steps. Each step involves looking at a specific operation and propagating between tensors (operands and results), based on the characteristics of that operation. Taking a matmul as an example, we would propagate between the non-contracting dimension of either lhs or rhs to the corresponding dimension of the result, or between the contracting dimension of the lhs and rhs.
+
+The characteristics of an operation determine the connection between corresponding dimensions in its inputs and outputs, and can be abstracted as a per op [sharding rule](#operation-sharding-rule).
```

Comment:
Is the link here working?

Reply:
It doesn't work for me, but I guess it will once this PR is merged

Reply:
Yes this should work with any markdown static site generator.